doc/Motivation.md

```diff
@@ -0,0 +1,108 @@
+Motivation
+==========
+
+In the last couple of years, I have seen many frameworks and libraries which
+either serialize or parse data or provide some kind of RMI. None of those
+solutions satisfied me. In the following paragraphs, I describe some of the
+usual approaches and explain why I don't like them. Note that those paragraphs
+just contain my personal opinion and motivation for this project. If you don't
+want to read my rambling, you may skip this section without missing anything
+important.
+
+First of all, I'm not particularly fond of code generation. Generated code is
+often opaque, hard to debug. Code generators are essentially yet another
+dependency, which you have to deal with (also in your build environment), albeit
+that dependency is usually fulfilled by the library/framework requiring it.
+I also have to admit that there are indeed cases where code generation is the
+right thing to do: when a formal, language independent specification exists,
+which must be fulfilled (e.g. protobuf). However, the generated code should be
+as unintrusive and small in size as possible.
+
+Round trip engineering is, in my opinion, a fulsome solution to the wrong
+problem most of the time. The mantra of model driven development is that
+generating repetitive code is faster and less error prone than writing  (read
+C&P'ing)it by hand every time, which undoubtedly is true. However, it is my
+personal opinion that, if you have to write much repetitive code, either you are
+doing it wrong, the API/framework author is doing it wrong or you are simply
+using the wrong programming language (read: you are doing it wrong).
+
+In that particular category, much personal hatred goes towards the eclipse
+foundation, which, as it seems, doesn't even consider the use-case of using
+their tools outside an over-bloated IDE. But then again, I avoid Java whenever I
+can, anyway.
+
+Some libraries (especially XML or JSON libraries) just present you a
+super-dynamic data-structure mirroring a generic AST of the language and require
+you to get/set the data yourself manually and take care of every single error
+case conceivable. You end up writing lots of repetitive code and potentially
+buggy or outright broken code, which, as explained above, should be avoided
+
+Some frameworks generate code for (de)serialization and require you to feed it
+every single piece of data manually. This may cause you to depend on a specific
+version of a specific code generator and, again, requires the user to write a
+lot of repetitive code.
+
+Then there are frameworks that either require you to add a super class to the
+thing you want to serialize or make use of code-annotations and compiler-plugins
```

Comment:
If they are not of a very general nature, they require either code-generation or compiler-plugins.